src/server/routes/entity/entity.tsx

```diff
@@ -1085,6 +1086,21 @@ export function handleCreateOrEditEntity(
 			let allEntities = [...otherEntities, mainEntity]
 				.filter(entity => entity.get('dataId') !== null);
 
+			// copy name to the edition group
+			if (!isNew && entityType === 'Edition' && body.copyToEG) {
+				const currentEditionGroup = currentEntity.editionGroup;
+				const entityDefaultAlias = body.aliases.find((alias) => alias.default);
+				if (currentEditionGroup.defaultAlias.name !== entityDefaultAlias.name) {
+					const tempBody = {aliases: [{...currentEditionGroup.defaultAlias, default: true, name: entityDefaultAlias.name}]};
```

Comment:
Reading this aliases array, I think if the EG before modification has multiple aliases, this tempBody will always be a different array of aliases, since it onyl contains one item.
You'll have to use the whole `currentEditionGroup.aliases` and modify the default alias (I guess by filtering by `id`).

I don't think we have access to the aliases array on currentEditionGroup and so the `const EGEntity = await fetchOrCreateMainEntity` call should probably be done beforehand.

Reply:
About that last detail: I've suggested removing the `currentEditionGroup.defaultAlias.name !== entityDefaultAlias.name` check and letting `getNextAliasSet` do the comparison, but now I realize this adds some DB calls if we have to fetch the Edition Group first to get the aliases (and we don't want that).

I *think* we can get away with fetching all the aliases for the EG in makeEntityLoader: https://github.com/metabrainz/bookbrainz-site/blob/master/src/server/routes/entity/edition.js#L299 
`'editionGroup.defaultAlias'` -> `'editionGroup.aliases'` probably does the trick?

If none of this works then we'll want to keep an alias comparison condition, but make it similar to what `updateAliasSet` ORM function uses: https://github.com/metabrainz/bookbrainz-data-js/blob/master/src/func/alias.ts#L41-L48